src/loadConfigurationFile.js

```diff
@@ -60,5 +60,10 @@ module.exports = function(configPath) {
             throw new Error('Could not load required module loading for ' + chalk.underline(configPath));
         }
     }
-    return require(configPath);
+    var req = require(configPath);
+    if (typeof req === 'function') {
+        return req();
```

Comment:
It's likely a bit more complicated but it should work like this:

1) Within bin/run.js collect all arguments that are prefixed with `env.` and pass them to the `run` function as part of the passed in object (choose `env` as property name)
2) Inside of `src/loadConfigurationFile.js` add a new parameter `envOptions` to the exported function (this place right here) and pass it in your line 65 into the function
3) Inside of `src/webpackWorker.js` pass `options.env` into the `loadConfigurationFile` function call as second parameter
4) Inside of `index.js`, pass `options.env` within the `run` function as second parameter to the `loadConfigurationFile` invocation

That should roughly cover it. Not entirely sure if `minimist` (the library we're using for dealing with command line arguments) can help you with identifying the `env.`-prefixed arguments though.

Let me know if you need anything. Thanks again!